purchase_order_sending_warning/models/mail_mail.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class MailMail(models.Model):
+    _inherit = "mail.mail"
+
+    def write(self, vals):
+        res = super(MailMail, self).write(vals)
+        if vals.get("state") == "exception":
+            for mail in self:
+                if mail.model == "purchase.order":
+                    order = self.env["purchase.order"].browse(mail.res_id)
+                    order.sending_warning = "email_not_sent"
+        else:
+            for mail in self:
+                if mail.model == "purchase.order":
+                    order = self.env["purchase.order"].browse(mail.res_id)
+                    order.sending_warning = False
```

Comment:
Maybe refactor like this:
```suggestion
        res = super().write(vals)
        po_mails = res.filtered_domain([("model", "=", "purchase.order")])
        if po_mails:
            error = vals.get("state") == "exception"
            records = self.env["purchase.order"].browse(po_mails.mapped("res_id"))
            records.write({"sending_warning": "email_not_sent" if error else False})
```